app/src/main/java/com/github/se/icebreakrr/model/message/MessagingTimeoutWorker.kt

```diff
@@ -0,0 +1,21 @@
+package com.github.se.icebreakrr.model.message
+
+import android.content.Context
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+
+/**
+ * This class is used to measure the time taken for a meeting request to be responded, if it isn't
+ * after 20 min, the meeting request is cancelled
+ */
+class MessagingTimeoutWorker(context: Context, params: WorkerParameters) : Worker(context, params) {
+  override fun doWork(): Result {
+    val meetingRequestViewModel = MeetingRequestManager.meetingRequestViewModel
+    val targetUID = inputData.getString("TARGET_UID") ?: "Unknown UID"
+    val targetToken = inputData.getString("TARGET_TOKEN") ?: "Unknown Key"
+    val targetName = inputData.getString("TARGET_NAME") ?: "Unknown name"
+    meetingRequestViewModel?.sendCancellationToBothUsers(
```

Comment:
You can send a message to yourself? Isn't it better to send the message to the user where this code is not running and update locally for yourself?

Reply:
Well, the thing is that the MeetingRequestService is the one to have access to the function to create notifications, therefore it is the only one who can call it. So it is way simpler and more clean to send a message to yourself with the handeling of the meetingRequestService than to do more boilerplate code to change it just locally directly. I think that like this it is cleaner and it works well

Reply:
If you think it is good like that go ahead.